packages/ui/src/primitives/tailwind/Select/index.tsx

```diff
@@ -23,181 +23,259 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
 Infinite Reality Engine. All Rights Reserved.
 */
 
-import React, { useEffect, useRef } from 'react'
-import { useTranslation } from 'react-i18next'
-import { MdOutlineKeyboardArrowDown } from 'react-icons/md'
-import { twMerge } from 'tailwind-merge'
-
-import { calculateAndApplyYOffset } from '@ir-engine/common/src/utils/offsets'
 import { useClickOutside } from '@ir-engine/common/src/utils/useClickOutside'
-import { useHookstate } from '@ir-engine/hyperflux'
+import { CheckLg, ChevronDownSm, HelpIconSm } from '@ir-engine/ui/src/icons'
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { InputProps, variantSizes } from '../Input'
+import Tooltip from '../Tooltip'
 
-import Input, { InputProps } from '../Input'
+export interface MenuItemProps {
+  children: React.ReactNode
+  value: string | number
+  disabled?: boolean
+  selected?: boolean
+  onClick?: () => void
+  showCheckmark?: boolean
+}
 
-export type OptionValueType = string | number
+export const MenuItem = ({ children, selected, disabled, onClick, showCheckmark }: MenuItemProps) => {
+  return (
+    <div
+      tabIndex={0}
+      className={twMerge(
+        `flex cursor-pointer items-center gap-x-2 bg-[#141619] px-4 py-2.5 text-xs text-[#9CA0AA] hover:bg-[#191B1F] hover:text-[#F5F5F5] ${
+          selected && 'text-[#375DAF]'
+        } ${disabled && 'text-[#42454D]'} w-full`
+      )}
+      onClick={onClick}
+    >
+      <div className="w-full">{children}</div>
 
-export type SelectOptionsType = { label: string; value: any; disabled?: boolean }
+      {selected && showCheckmark && <CheckLg color="#375DAF" />}
+    </div>
+  )
+}
 
-export interface SelectProps<T extends OptionValueType> {
-  label?: string
-  className?: string
-  error?: string
-  description?: string
-  options: SelectOptionsType[]
-  currentValue: T
+export interface SelectProps<T = string | number> {
+  children: React.ReactElement<typeof MenuItem> | React.ReactElement<typeof MenuItem>[]
+  width?: 'sm' | 'md' | 'lg' | 'full'
+  inputSizeVariant?: InputProps['variantSize']
   onChange: (value: T) => void
-  placeholder?: string
+  value: T
+  renderValue?: (value: T) => React.ReactNode
+  labelProps?: InputProps['labelProps']
+  state?: InputProps['state']
+  helperText?: InputProps['helperText']
+  required?: boolean
+  showCheckmark?: boolean
```

Comment:
we should show the checkmark by default
and hide it when needed

Reply:
Removed this prop since all designs are using it.